hibernate-reactive-core/src/main/java/org/hibernate/reactive/adaptor/impl/QueryParametersAdaptor.java

```diff
@@ -7,31 +7,53 @@
 
 import org.hibernate.JDBCException;
 import org.hibernate.engine.spi.QueryParameters;
-import org.hibernate.engine.spi.SessionImplementor;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
 import org.hibernate.type.Type;
 
+import java.io.Serializable;
+import java.sql.PreparedStatement;
 import java.sql.SQLException;
 
 public class QueryParametersAdaptor {
 
-	public static Object[] toParameterArray(QueryParameters queryParameters, SessionImplementor session) {
+	public static Object[] toIdParameterArray(
+			final Serializable id,
+			final Type type,
+			final SharedSessionContractImplementor session) {
+		return toParameterArray( preparedStatement -> type.nullSafeSet( preparedStatement, id, 1, session ) );
+	}
+
+
+	public static Object[] toParameterArray(QueryParameters queryParameters, SharedSessionContractImplementor session) {
+
+		return toParameterArray(
+				preparedStatement -> {
+					Type[] types = queryParameters.getPositionalParameterTypes();
+					Object[] values = queryParameters.getPositionalParameterValues();
+					int n = 1;
+					for (int i = 0; i < types.length; i++) {
+						Type type = types[i];
+						Object value = values[i];
+							type.nullSafeSet(preparedStatement, value, n, session);
+							n += type.getColumnSpan(session.getFactory());
+					}
+				}
+		);
+	}
+
+	public static Object[] toParameterArray(Binder binder) {
 		PreparedStatementAdaptor adaptor = new PreparedStatementAdaptor();
-		Type[] types = queryParameters.getPositionalParameterTypes();
-		Object[] values = queryParameters.getPositionalParameterValues();
-		int n = 1;
-		for (int i = 0; i < types.length; i++) {
-			Type type = types[i];
-			Object value = values[i];
-			try {
-				type.nullSafeSet(adaptor, value, n, session);
-				n += type.getColumnSpan( session.getSessionFactory() );
-			}
-			catch (SQLException e) {
-				//can never happen
-				throw new JDBCException("error binding parameters", e);
-			}
+		try {
+			binder.bind( adaptor );
+			return adaptor.getParametersAsArray();
+		}
+		catch (SQLException e) {
+			//can never happen
+			throw new JDBCException("error binding parameters", e);
 		}
-		return adaptor.getParametersAsArray();
 	}
```

Comment:
@gbadner I'm wondering if this static method would work better as a constructor of `PreparedStatementAdaptor`. WDYT?

Reply:
Actually on second thoughts: I tried inlining the method and HR didn't get worse. It's not really doing much except for the exception translation. I personally would get rid of it. (This is very minor though.)